qiskit/compiler/transpile_config.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2019, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""Models for TranspileConfig and its related components."""
+
+from qiskit.validation import BaseModel, BaseSchema, bind_schema
+
+
+class TranspileConfigSchema(BaseSchema):
```

Comment:
empty schemas and models?

Reply:
yeah, I would like to add it but I want to work out which ones. 

basis_gates, coupling_map. 

Not backend which we do while the transpiler needs a backend.

Reply:
I think the fields should be:
```basis_gates, coupling_map, initial_layout, seed_mapper```
all optional. and we can extend later.